### ISSUE TITLE
Early stop does not work

### ISSUE THREAD

**Issue by mahnehsilla** [tensorflow/tfjs#7473]
* Tracking `"acc"` won't stop using `tf.callbacks.earlyStopping` even after several times with the same value.

* This is a very simple MWE that you can open with live server:

* Should be fixed, this is very useful !!

<details>

<summary> <b>SEE CODE</b> </summary>
```
<!DOCTYPE html>
<html lang="en">
  <script
    src="https://cdnjs.cloudflare.com/ajax/libs/tensorflow/4.2.0/tf.min.js"
    integrity="sha512-luqeEXU5+ipFs8VSUJZTbt6Iil1m7OT0bODSccqew2CN85iad5Mn//M9+CPVI4UGlo8kN51OWFSox+fYe4qgYQ=="
    crossorigin="anonymous"
    referrerpolicy="no-referrer"
  ></script>
  <body>
    <script>
      function makeModel(units = 1, inputShape = [1], lr = 0.1) {
        const seq = tf.sequential();
        seq.add(tf.layers.dense({ units, inputShape }));
        seq.compile({
          loss: "meanSquaredError",
          optimizer: tf.train.adam(lr),
          metrics: ["acc"],
        });
        seq.summary();
        return seq;
      }

      async function doTraining(model, xs, ys) {
        const history = await model.fit(xs, ys, {
          epochs: 500,
          callbacks: {
            onEpochEnd: async (epoch, loss) => {
              console.log("Epoch: ", epoch, "Loss: ", loss);
            },
            onEarlyStop: tf.callbacks.earlyStopping({ monitor: "acc" }),
          },
        });
        console.log(history);
        return model;
      }

      let xs = tf.range(0, 200);
      xs = xs.div(xs.shape[0]);
      let ys = tf.range(40, 240);
      ys = ys.div(ys.shape[0]);

      const seq = makeModel(1, 1, 0.01);

      const input = document.getElementById("input");
      const info = document.getElementById("information");

      doTraining(seq, xs, ys).then((model) => {
        console.log("DONE");
      });
    </script>
  </body>
</html>
```

</details>

**Comment by gaikwadrahul8** [comment]
Hi, @mahnehsilla 

Apologize for the delayed response and I was able to replicate the same issue from my end also, For your reference I have added codepen[ link](https://codepen.io/gaikwadrahul8/pen/MWqVwzR?editors=0012&layout=top), I observed this unexpected behaviour from `tf.callbacks.earlyStopping` while monitoring on the accuracy because accuracy is not changing  or increasing so ideally `tf.callbacks.earlyStopping` should stop training early but in this case, It's not happening at the moment so we'll have to dig more into this issue and we'll update you soon. Thank you for noticing this issue. I really appreciate it. Thank you!

**Comment by mahnehsilla** [comment]
No problem, it would be great if you can fix it. Thank you.

**Comment by gaikwadrahul8** [comment]
Hi, @linchenn

Could you please look into this issue? Thank you!

**Comment by KSANTHOSH200** [comment]
Hey, 

Can I work on this?

Thanks.

**Comment by Linchenn** [comment]
@gaikwadrahul8 Thank you for reproducing it!

@KSANTHOSH200 Sure, it would be appreciated! Before coding, could you share your ideas about the problem and how would you fix it?

**Comment by sergiuionescu** [comment]
@mahnehsilla Instead of passing a list of callbacks you are passing an object containing a callback function and another object which seems to override the onEpochEnd callback of earlyStopping.

This should work and keep both your custom callback and earlyStopping:

```
async function doTraining(model, xs, ys) {
    const history = await model.fit(xs, ys, {
      epochs: 500,
      callbacks: [
        new tf.CustomCallback({onEpochEnd: async (epoch, loss) => {
          console.log("Epoch: ", epoch, "Loss: ", loss);
        }}),
        tf.callbacks.earlyStopping({ monitor: "acc" }),
      ],
    });
    console.log(history);
    return model;
  }

  let xs = tf.range(0, 200);
  xs = xs.div(xs.shape[0]);
  let ys = tf.range(40, 240);
  ys = ys.div(ys.shape[0]);

  const seq = makeModel(1, 1, 0.01);

  const input = document.getElementById("input");
  const info = document.getElementById("information");

  doTraining(seq, xs, ys).then((model) => {
    console.log("DONE");
  });
```

See a similar issue reported here:
https://stackoverflow.com/questions/58116087/tensorflowjs-with-earlystopping-and-training-logs-does-not-work/65456805#65456805

**Comment by gaikwadrahul8** [comment]
Hi, @mahnehsilla 

I tried the workaround suggested by @sergiuionescu and it seems like working as expected for your reference I have added modified codepen example [here](https://codepen.io/gaikwadrahul8/pen/MWqVwzR?editors=1112) so please try it from your end and if it's working as expected please feel free to close this issue.

Thank you for your understanding and patience.
